qtpy/uic.py

```diff
@@ -81,9 +81,11 @@
     if PYSIDE:
         from PySide.QtCore import QMetaObject
         from PySide.QtUiTools import QUiLoader
+        from pysideuic import compileUi
     elif PYSIDE2:
         from PySide2.QtCore import QMetaObject
         from PySide2.QtUiTools import QUiLoader
+        from pyside2uic import compileUi
```

Comment:
Sorry if this is a stupid question, but where do I get this `pyside2uic` module?

Got failing tests at the moment because of this missing module:
```
qtpy/uic.py:88: in <module>
from pyside2uic import compileUi
/usr/lib/python3.9/site-packages/shiboken2/files.dir/shibokensupport/__feature__.py:142: in _import
return original_import(name, *args, **kwargs)
E   ModuleNotFoundError: No module named 'pyside2uic'
```

I can't find this module in the sources for PySide2(-tools) (or at least not in the sources for version 5.15.2)

Reply:
https://github.com/pyside/pyside2-tools

Reply:
> https://github.com/pyside/pyside2-tools

Thanks, but I'm still a bit confused. That GitHub repo is quite old, and after digging a bit deeper into this it seems that more recent versions of pyside2-tools have removed "pyside2-uic and pyside2-rcc in favor of using Qt's uic and rcc tools directly with the `-g python`option." [1]

So does that mean that to package QtPy properly I have to package this old version of pyside2-tools as well and depend on that? To be honest it seems far from ideal to me to force such a downgrade through dependency resolution, especially considering that mixing different versions of pyside2/shiboken2/pyside2-tools is probably a recipe for disaster. Pyside2 already breaks when shiboken2 was compiled with a different _minor_ version of clang, so I'm hesitant to start mixing version numbers within PySide2 and friends. 

How do I resolve this?

[1] https://code.qt.io/cgit/pyside/pyside-tools.git/commit/?h=v5.15.2&id=c976560a1a8fef26f117b7aec2f60358a7ef91b6

Reply:
> So does that mean that to package QtPy properly I have to package this old version of pyside2-tools as well and depend on that?

No, it doesn't. This only be used if `pyside2uic` is found. That's why the test is skipped for PySide2, so please don't run it and you'll  be good.

Reply:
Or you could help us to improve this by changing `pyside2uic` to `uic -g python`.

Reply:
> > So does that mean that to package QtPy properly I have to package this old version of pyside2-tools as well and depend on that?
> 
> No, it doesn't. This only be used if `pyside2uic` is found. That's why the test is skipped for PySide2, so please don't run it and you'll be good.

Alright, thanks for explaining. 

The problem however is that even though the test should be skipped, `test_uic.py` still imports `uic.py` which tries to import `pyside2uic` if PYSIDE2:
```
python3.8 -m pytest -vv -ra -l -Wdefault
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/python3.8
cachedir: .pytest_cache
rootdir: /var/tmp/portage/dev-python/QtPy-1.11.0/work/QtPy-1.11.0
plugins: pkgcore-0.12.7
collecting ... collected 42 items / 1 error / 41 selected

====================================================================================== ERRORS ======================================================================================
_____________________________________________________________________ ERROR collecting qtpy/tests/test_uic.py ______________________________________________________________________
ImportError while importing test module '/var/tmp/portage/dev-python/QtPy-1.11.0/work/QtPy-1.11.0/qtpy/tests/test_uic.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/site-packages/_pytest/python.py:578: in _importtestmodule
mod = import_path(self.fspath, mode=importmode)
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:524: in import_path
importlib.import_module(module_name)
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
???
<frozen importlib._bootstrap>:991: in _find_and_load
???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
???
<frozen importlib._bootstrap>:671: in _load_unlocked
???
/usr/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
exec(co, module.__dict__)
qtpy/tests/test_uic.py:8: in <module>
from qtpy import uic
/usr/lib/python3.8/site-packages/shiboken2/files.dir/shibokensupport/__feature__.py:142: in _import
return original_import(name, *args, **kwargs)
qtpy/uic.py:88: in <module>
from pyside2uic import compileUi
/usr/lib/python3.8/site-packages/shiboken2/files.dir/shibokensupport/__feature__.py:142: in _import
return original_import(name, *args, **kwargs)
E   ModuleNotFoundError: No module named 'pyside2uic'
============================================================================= short test summary info ==============================================================================
ERROR qtpy/tests/test_uic.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================= 1 error in 0.49s =================================================================================
* pytest failed with python3.8
```

> Or you could help us to improve this by changing pyside2uic to uic -g python.

Would love to help out, but I'm afraid I lack the knowledge require to do so. I'll look into fixing the skipping of this test though.

Reply:
> I'll look into fixing the skipping of this test though.

See: https://github.com/spyder-ide/qtpy/pull/244